sw/host/provisioning/ft_lib/src/lib.rs

```diff
@@ -159,6 +163,66 @@ pub fn test_exit(
     Ok(())
 }
 
+// Execute an openssl invocation, passing the args[] as command line parameters.
+// The intended use is openssl x509 certificate verification. cert_num is the
+// number of the certificate in the list of certificates being validated.
+fn openssl_command(cert_num: usize, args: &[&str]) -> Result<()> {
+    let o = Command::new("openssl").args(args).output()?;
+    if !o.status.success() {
+        log::error!(
+            "openssl output:\n{}",
+            std::str::from_utf8(&o.stderr).unwrap()
+        );
+        Err(anyhow!("Cert #{cert_num}: {} failed", args[0]))
+    } else {
+        Ok(())
+    }
+}
+
+// Validate the passed in certificates using 'openssl verify ...' command.
+// ca_pem is the file name of the CA certificate saved in PEM format. cers is a
+// vector of certificate binary blobs.
+fn validate_certs_chain(ca_pem: &str, certs: &[&Vec<u8>; 3]) -> Result<()> {
+    let base_name = tmpfilename("cert_validation");
+    let binding_der = base_name.to_owned() + ".der";
+    let binding_pem = base_name.to_owned() + ".pem";
+
+    let der_filename = binding_der.as_str();
+    let pem_filename = binding_pem.as_str();
+
+    for (i, cert) in certs.iter().enumerate() {
+        let size = (u16::from_be_bytes([cert[2], cert[3]]) + 4) as usize;
```

Comment:
here you should have at least some sanity checks like DER encoding and also verify that the size is not too large

Reply:
done